components/service-operator/controllers/sqs_cloudformation_test.go

```diff
@@ -0,0 +1,170 @@
+					Labels: map[string]string{
+						cloudformation.AccessGroupLabel: "test.access.group",
+					},
+				},
+				Spec: queue.SQSSpec{
+					Secret: secretName,
+				},
+			}
+			secret core.Secret
+		)
+
+		By("creating a prequisite Principal resource with kubernetes api", func() {
+			Expect(client.Create(ctx, &principal)).To(Succeed())
+		})
+
+		By("creating an SQS resource with kubernetes api", func() {
+			Expect(client.Create(ctx, &sqs)).To(Succeed())
+		})
+
+		By("displaying a READY resource status after initial creation", func() {
+			Eventually(func() object.State {
+				_ = client.Get(ctx, resourceNamespacedName, &sqs)
+				return sqs.GetState()
+			}, timeout).Should(Equal(object.ReadyState))
+		})
+
+		By("displaying an AWS CREATE_COMPLETE resource status after initial creation", func() {
+			Eventually(func() string {
+				_ = client.Get(ctx, resourceNamespacedName, &sqs)
+				return sqs.Status.AWS.Status
+			}, timeout).Should(Equal(cloudformation.CreateComplete))
+		})
+
+		By("displaying an AWS stack id in resource status", func() {
+			Eventually(func() string {
+				_ = client.Get(ctx, resourceNamespacedName, &sqs)
+				return sqs.Status.AWS.ID
+			}).ShouldNot(BeEmpty())
+		})
+
+		By("displaying a stack name prefixed with cluster name in resource status", func() {
+			Eventually(func() string {
+				_ = client.Get(ctx, resourceNamespacedName, &sqs)
+				return sqs.Status.AWS.Name
+			}).Should(ContainSubstring("xxx-sqs-test-test-queue"))
+		})
+
+		By("ensuring a finalizaer is present on resource to prevent deletion", func() {
+			Eventually(func() []string {
+				_ = client.Get(ctx, resourceNamespacedName, &sqs)
+				return sqs.Finalizers
+			}).Should(ContainElement(cloudformation.Finalizer))
+		})
+
+		By("ensuring no DeletionTimestamp exists", func() {
+			Eventually(func() bool {
+				_ = client.Get(ctx, resourceNamespacedName, &sqs)
+				return sqs.ObjectMeta.DeletionTimestamp == nil
+			}).Should(BeTrue())
+		})
+
+		By("creating a secret with queue connection details", func() {
+			Eventually(func() map[string][]byte {
+				_ = client.Get(ctx, secretNamespacedName, &secret)
+				return secret.Data
+			}).Should(HaveKey("QueueURL"))
+		})
+
+		By("creating a secret with the prinical role name", func() {
+			Eventually(func() map[string][]byte {
+				_ = client.Get(ctx, secretNamespacedName, &secret)
+				return secret.Data
+			}).Should(HaveKey("IAMRoleName"))
+		})
+
+		By("deleteing SQS resource with kubernetes api", func() {
+			err := client.Get(ctx, resourceNamespacedName, &sqs)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(client.Delete(ctx, &sqs)).To(Succeed())
+		})
+
+		By("ensuring the SQS resources have been removed", func() {
+			Eventually(func() int {
+				var list queue.SQSList
+				err := client.List(ctx, &list)
+				Expect(err).ToNot(HaveOccurred())
+				return len(list.Items)
+			}, timeout).Should(Equal(0))
+		})
+
+		// By("ensuring secret has been removed", func() {
```

Comment:
comment from the others: `// GC will remove this in a real cluster, but we don't have the hooks installed in our tests :(`